### PR TITLE
In case of any Throwable inside a test mark the test as failed.

### PR DESCRIPTION
**Pull request by mzuehlke**
closes: #650

### CODE REVIEW

File: munit/shared/src/main/scala/munit/MUnitRunner.scala [hunk 1]
```diff
@@ -310,6 +310,9 @@ class MUnitRunner(val cls: Class[_ <: Suite], newInstance: () => Suite)
         handleNonFatalOrStackOverflow(ex)
       case ex: StackOverflowError =>
         handleNonFatalOrStackOverflow(ex)
+      case ex =>
```
Comment (filipwiech) [inline munit/shared/src/main/scala/munit/MUnitRunner.scala:313]: If this is a fatal exception, would it make sense to abort the suite then, or not really/it doesn't matter? :thinking:

Comment (mzuehlke) [reply munit/shared/src/main/scala/munit/MUnitRunner.scala:313]: Very good idea 👍

**Comment by mzuehlke** [comment]
@tgodzik  do you have an opinion about this pull request.

**Review by tgodzik — APPROVED** [review]
Look good to me though I haven't touched the codebase too much

**Pull request merged**